File: Python3/프로그래머스/0/181897. 리스트 자르기/리스트 자르기.py
def solution(n, slicer, num_list):
    a,b,c = slicer[0], slicer[1] + 1, slicer[2]
    if n == 2:
        b = len(num_list)
    if n == 1:
        a = 0
    if n != 4 :
        c = 1
        
    return num_list[a:b:c]

File: C++/프로그래머스/0/181897. 리스트 자르기/리스트 자르기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n, vector<int> slicer, vector<int> num_list) {
    if(n == 1) slicer[0] = 0;
    if(n == 2) slicer[1] = num_list.size() - 1;
    if(n != 4) slicer[2] = 1;
    
    vector<int> answer;
    for(int i = slicer[0]; i <= slicer[1]; i+=slicer[2]) answer.push_back(num_list[i]);
    
    return answer;
}